### ISSUE TITLE
CLI exits with code 2 after logging only warnings when the styled-components processor is used

### ISSUE THREAD

**Issue by ismay** [stylelint/stylelint#2473]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Stylelint exits with exit code 2 for a run with only warnings (see [this build](https://travis-ci.org/ismay/ismaywolff.nl/builds/219376502#L560)). As I see it, it should exit with exit code 0, after logging the warnings, according to [the](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/cli.md#exit-codes) [docs](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#severities-error--warning).

> Which rule, if any, is this issue related to?

The [stylelint-no-unsupported-browser-features](https://github.com/ismay/stylelint-no-unsupported-browser-features) plugin, and the accompanying `plugin/no-unsupported-browser-features` rule.

> What CSS is needed to reproduce this issue?

Just anything that will cause doiuse to complain. So this css:

```css
.foo {
  width: 10rem;
}
```

With this browserslist:

```
 > 5%
  Last 2 versions
```

is enough

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "processors": [
    "stylelint-processor-styled-components"
  ],
  "plugins": [
    "stylelint-no-unsupported-browser-features"
  ],
  "extends": "stylelint-config-standard",
  "syntax": "scss",
  "rules": {
    "rule-empty-line-before": null,
    "plugin/no-unsupported-browser-features": [true, { "severity": "warning" }]
  }
}
```

> Which version of stylelint are you using?

`7.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint *.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, scss syntax with `stylelint-processor-styled-components`

> What did you expect to happen?

An exit code of 0

> What actually happened (e.g. what warnings or errors you are getting)?

Exit code 2 (see the build above)

**Comment by ismay** [comment]
After testing I suspect that this is related to `stylelint-processor-styled-components`. After removing that from the chain the process exits with 0. I'll open there, this can be closed as it doesn't seem to be directly stylelint related.

**Comment by jeddy3** [comment]
@ismay Thanks for the detailed report. I suspect this will need a bit more digging. Feel free to help out by poking around the processor code in stylelint to see if something is amiss with the severity handling.

**Comment by ismay** [comment]
> Feel free to help out by poking around the processor code in stylelint to see if something is amiss with the severity handling.

I'm planning on moving my app to https://github.com/paypal/glamorous, since styled-components is too heavy for my use-case. Glamorous'll need a stylelint-processor as well, so I might be basing that on the one for styled-components. If I encounter anything there I'll let you know (feel free to ping me about it if it takes too long, I might forget :).

**Comment by jeddy3** [comment]
> Glamorous'll need a stylelint-processor as well, so I might be basing that on the one for styled-components.

Good stuff! Please drop us a PR to add it to the [list of processsors](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/processors.md) if you publish it.

>  If I encounter anything there I'll let you know

Thanks.

**Comment by ismay** [comment]
So after a bit of a detour I revisited this. I've narrowed it down to `stylelint-config-standard`. As soon as I add that to my config (see [here](https://github.com/ismay/ismaywolff.nl/pull/325)), the cli exits with status code 2, without logging any errors:

* [Failing travis build (with stylelint-config-standard)](https://travis-ci.org/ismay/ismaywolff.nl/builds/222325474#L785)
* [Successful build (without stylelint-config-standard)](https://travis-ci.org/ismay/ismaywolff.nl/builds/222323733#L781)

**Comment by ismay** [comment]
@evilebottnawi Try this pr: https://github.com/ismay/ismaywolff.nl/pull/325, before these changes the build was succeeding. I've been trying to reproduce with a smaller codebase, but haven't been successful so far.

And you don't need glamorous, I'm using styled-components. My stylelint config for the linked pr is:

```json
{
  "extends": "stylelint-config-standard",
  "processors": ["stylelint-processor-styled-components"],
  "syntax": "scss",
  "plugins": [
    "stylelint-no-unsupported-browser-features"
  ],
  "rules": {
    "rule-empty-line-before": null,
    "plugin/no-unsupported-browser-features": [true, { "severity": "warning" }]
  }
}
```

**Comment by ismay** [comment]
@evilebottnawi So I've narrowed it down a bit. This is the reduced test case: 

* Repo: https://github.com/ismay/stylelint-test
* Failing build: https://travis-ci.org/ismay/stylelint-test/builds/222335661#L202

**Comment by evilebottnawi** [comment]
@ismay if you change to `"test": "stylelint 'jsx/Spinner.jsx'"` all works fine, if you use `"test": "stylelint 'jsx/*.jsx'"` you get `Routes.test.jsx` and `Spinner.jsx`, seems bug in `stylelint-processor-styled-components`

**Comment by evilebottnawi** [comment]
@ismay you can use `"test": "stylelint 'jsx/*.jsx' '!**/*.test.jsx'"` to solve local problems

**Comment by evilebottnawi** [comment]
@ismay see https://github.com/styled-components/stylelint-processor-styled-components/pull/23, just disable `no-empty-source` and `no-missing-end-of-source-newline` rules, it is the best solution. Unfortunately in this case this does not apply to `stylelint` and we can not do anything here :sob: 
.

**Comment by ismay** [comment]
Ah great! Ignoring `no-empty-source` and `no-missing-end-of-source-newline` fixed it. Awesome that you submitted a pr with styled-components to fix this as well 👍 Thanks for the help @evilebottnawi!
